### ISSUE TITLE
react-apollo swallows render errors of wrapped components 

### ISSUE THREAD

**Issue by bessey** [apollostack/react-apollo#347]
## Steps to Reproduce

My case is produced as follows:

- Have a component that will raise an error in its `render()`, when a query is complete e.g. 
```jsx
render () {
  let {loading} = this.props.data 
  return (
    { loading ? null : <RealComponent blah={this.thisMethodIsntReal()} /> }
  )
}
```
- Wrap it with the `graphql` HOC (`import { graphql } from 'react-apollo'`)

Note that I haven't tried simplifying this test case and its possible less setup is required.

### Buggy Behavior

Rather than raising `this.thisMethodIsntReal is not a function`, Apollo catches the error, and passes it into the component as `data`, wiping out the actual data returned from the query. In doing so it (likely) causes a different obscure error in your component, because the component depended on the now missing data.

Here's my example with a `debugger` in `render()` to show the contents of `this.props.data`

![image](https://cloud.githubusercontent.com/assets/708200/20729291/39069664-b636-11e6-8712-555acaa944a1.png)

### Expected Behavior

Apollo should not catch errors it did not cause, this error should propagate to the console and so the second state, with no data, should never be reached.

### Version
- apollo-client@0.5.7
- react-apollo@0.6.0

**Comment by tmeasday** [comment]
Thanks @bessey. This is an issue a few have run into, thanks for the simple repro.

**Comment by bessey** [comment]
This is my current workaround, which may be helpful to others. I wrap components in a HOC that doesn't swallow the error so its `apollo(throwApolloError(component))`

https://gist.github.com/bessey/926cdb3e383b406e7f546c6f4dd55a8a

**Comment by tmeasday** [comment]
I think this should be fixed by `apollo-client@0.5.9`. Can you give it a try and let me know?

**Comment by bessey** [comment]
It is indeed. Thanks!
